src/Security/Authentication/Certificate/samples/Certificate.Sample/Startup.cs

```diff
@@ -37,6 +37,8 @@ public void ConfigureServices(IServiceCollection services)
                     };
                 });
 
+            // Adding a ICertificateValidationCache will result in certificate auth caching the results, the default implementation uses a memory cache
+            services.AddSingleton<ICertificateValidationCache, CertificateValidationCache>();
```

Comment:
Should we add an extension for this on the auth builder for discovery?

Reply:
like service.AddCertificate().AddDefaultCache()?

Reply:
Yeah,
```
services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme)
                .AddCertificate(...)
                .AddCertificateCache(options => {})
```